### ISSUE TITLE
Update Brave Talk URL in Sidebar to talk.brave.com

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#16709]
## Description 

Update Brave Talk URL in Sidebar to `talk.brave.com`; once https://github.com/brave/devops/issues/5168 lands, and creates an `HTTP 301 Permanent Redirect` from `together.brave.com` -> `talk.brave.com`, I think the following code might break Brave Talk in the Sidebar:

https://github.com/brave/brave-core/blob/ea66a8c49ade46d0d3d892626e7eea0948a95e09/components/sidebar/sidebar_service.cc#L36

https://github.com/brave/brave-core/blob/d9d0401c82d99284af60f7c81779aa632a136286/browser/ui/views/sidebar/sidebar_items_contents_view.cc#L393

<img width="1312" alt="Screen Shot 2021-06-30 at 9 30 19 AM" src="https://user-images.githubusercontent.com/387249/123999857-d0185500-d987-11eb-99ba-a7fc272de16c.png">

## Steps to Reproduce

   1. after https://github.com/brave/devops/issues/5168 lands:
   2. enable Sidebar via `brave://flags` and restart Brave
   2. click on the video-camera icon in the Sidebar for Brave Talk
   3. ensure it still loads and launches `talk.brave.com` via its `Start video call` button

## Reproduces how often: 

Unsure; this is pre-emptively filed to try to avoid/mitigate breakage on `nightly`.

## Brave version (brave://version info)

Brave | 1.28.37 Chromium: 91.0.4472.124 (Official Build) nightly (x86_64)
-- | --
Revision | 7345a6d1bfcaff81162a957e9b7d52649fe2ac38-refs/branch-heads/4472_114@{#6}
OS | macOS Version 11.4 (Build 20F71)

/cc @rebron @simonhong @mattmcalister @mrose17 casting the net wide; trying to get our in-product renaming done so there are no late surprises 👍

**Comment by bsclifton** [comment]
Reverted with https://github.com/brave/brave-core/pull/9347

**Comment by simonhong** [comment]
We saw crash https://github.com/brave/brave-browser/issues/16787 after changing default item's url.

**Comment by simonhong** [comment]
@stephendonner As we reverted https://github.com/stephendonner/brave-core/commit/c323b2ea7996697dc6681424c9e82239f92719d8, this issue is still not resolved.
I pushed another PR (https://github.com/brave/brave-core/pull/9349) for fixing this.

**Comment by simonhong** [comment]
Yes, it's strange to me, too. how this is automatically closed by https://github.com/stephendonner/brave-core/commit/c323b2ea7996697dc6681424c9e82239f92719d8 :)

**Comment by stephendonner** [comment]
Verified `FIXED` using 


Brave | 1.29.20 Chromium: 92.0.4515.107 (Official Build) nightly (x86_64)
-- | --
Revision | 87a818b10553a07434ea9e2b6dccf3cbe7895134-refs/branch-heads/4515@{#1634}
OS | macOS Version 12.0 (Build 21A5284e)

(Although I made the earlier patch, it got backed out, so I feel it's ok to verify @simonhong's fix.)


Steps:
1. new profile
2. launched Brave
3. enabled Sidebar via `brave://flags`
4. relaunched
5. launched Charles Proxy and listened for traffic
6. clicked on the `Brave Talk` menu item in the sidebar
7. clicked on `Start call`

Confirmed that traffic went to `talk.brave.com`, as expected.

<img width="1298" alt="Screen Shot 2021-07-22 at 5 23 07 PM" src="https://user-images.githubusercontent.com/387249/126724748-1f4f06d0-1e24-4636-a90f-5a3837fea372.png">
